lib/util/player_configuration.js

```diff
@@ -74,6 +74,9 @@ shaka.util.PlayerConfiguration = class {
       clearKeys: {},  // key is arbitrary key system ID, value must be string
       advanced: {},    // key is arbitrary key system ID, value is a record type
       delayLicenseRequestUntilPlayed: false,
+      initDataTransform: (initData, initDataType, drmInfo) => {
```

Comment:
@joeyparrish It seems that the test fails due to unused params, but the params are necessary in some cases, but not by default, how can resolve it?

Reply:
Let me see if I can satisfy the compiler and contribute an edit to this PR.  There's a comment at the top of this file with some background:

> Many times in our configs, we need to create an empty implementation of a method, but to avoid closure from removing unused parameters (and breaking our merge config code) we need to use each parameter. Is there a better solution to this problem than what we are doing now?

Reply:
I believe I have an answer to that ancient comment's question: "Is there a better solution to this problem than what we are doing now?"

I'll put up a PR with a new utility to solve this, and assuming you're already offline for the day, I'll try to rebase your PR for you and patch in the use of the new utility.

Reply:
I can do it this night (in 3-4hours)

Reply:
Just did it!